smartsim/_core/entrypoints/file_operations.py

```diff
@@ -0,0 +1,272 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024 Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import base64
+import collections
+import os
+import pickle
+import re
+import shutil
+from ast import literal_eval
+from distutils import dir_util  # pylint: disable=deprecated-module
```

Comment:
changed to shutil.copytree. Although I set ```dirs_exist_ok=True``` to make it work with my tests.

From the documentation: "If dirs_exist_ok is false (the default) and dst already exists, a [FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) is raised. If dirs_exist_ok is true, the copying operation will continue if it encounters existing directories, and files within the dst tree will be overwritten by corresponding files from the src tree."

Is that functionality okay?

Reply:
I'd actually prefer the behavior of `dirs_exist_ok=False` just so that we do not end up accidentally overwriting anything! This might be something worth getting the whole group's opinion on though.

Reply:
changed to False (error will be raised if the directory you want to copy into already exists)